imp/TokenSet.cs

```diff
@@ -176,18 +176,36 @@ private TokenSet()
         } /* end TokenSet */
 
         /* ---------------------------------------------------------------------------
-        * private constructor TokenSet (uint[])
-        * ---------------------------------------------------------------------------
-        * used to instantiate the prefabricated first and follow lists
+         * private TokenSet newFromRawData(uint[])
+         * ---------------------------------------------------------------------------
+         * Used to instantiate the prefabricated first and follow lists.
+         * Expects TSB[segmentCount] to be the number of tokens, and returns null if 
+         * the count does not match.
          * ------------------------------------------------------------------------ */
 
-        public TokenSet(params uint[] TSB)
+        public static TokenSet newFromRawData(params uint[] TSB)
         {
-            dataStored.segments[0] = TSB[0];
-            dataStored.segments[1] = TSB[1];
-            dataStored.segments[2] = TSB[2]; 
-            dataStored.elemCount = TSB[3];
-        }
+            TokenSet newSet = new TokenSet();
+            uint counter = 0;
+
+            for (int i = 0; i < segmentCount; i++) {
+
+                for (int j = 0; j < 32; j++) { 

```

Comment:
This loop variant is an index to address individual bits, a self explanatory name would be `bitIndex`.